### ISSUE TITLE
[MacOS] IsarError: Cannot open Environment: MdbxError (15): Block device required

### ISSUE THREAD

**Issue by jackstefansky** [isar/isar#1213]
I'm opening the same issue again ( see #1125 for more context ), since v3.1.0+1 does not fix the problem.

---

 - [x] I searched for similar issues already
 - [x] I filled the details section with the exact device model and version
 - [x] I am able to provide a reproducible example

**Comment by simc** [comment]
@jackstefansky sorry for incorrectly closing.


do you have more details about the issue?

Does it only happen on specific derices? Are you able to reproduce it? Any other details?

Did you also get this "error 15" in the previous issue?

**Comment by jackstefansky** [comment]
Yes, the #1125 was all about the error 15. I don't understand what does the "Block device required" means. 
When I started looking into it, I found some issues related to Mdbx database:

https://bytemeta.vip/repo/ledgerwatch/erigon/issues/2761
https://github.com/ledgerwatch/erigon/issues/4871

**Comment by jackstefansky** [comment]
Here are more info about the device that have this problem:

Model: MacBookPro16,1 
Architecture: x86 ( Intel )
Kernel version: Darwin Kernel Version 22.4.0: Mon Mar  6 21:00:17 PST 2023; root:xnu-8796.101.5~3/RELEASE_X86_64
Operating system version: 13.3.1(22E261)

**Comment by bartosz-timecamp** [comment]
@simc I'm experiencing the same issue on my macOS device with an Intel processor. Would you be able to provide any guidance on how to solve this problem, or suggest any potential workarounds? Thank you!

**Comment by jackstefansky** [comment]
@simc any clues what might cause this?

**Comment by turimm** [comment]
Unfortunately has the same issue:

MacOS Ventura 13.4.1 (22F82)
Macbook Pro 16,1
Intel Core i7

Flutter 3.10.5
Dart 3.0.5
Isar v3.1.0+1

**Comment by JulienDev** [comment]
@simc, I've also been observing a significant number of similar crashes recently. It appears to be predominantly affecting older Macbook models, however, it's not exclusively confined to them as I see some M2's on the list as well.

<img width="482" alt="image" src="https://github.com/isar/isar/assets/71345/9663d10f-4f2f-4428-8c0e-2b945ac13c82">
<img width="479" alt="Screenshot 2023-07-13 at 08 59 13" src="https://github.com/isar/isar/assets/71345/4ed7cc6f-8499-4783-bed5-3f45a9bca3a9">
<img width="477" alt="image" src="https://github.com/isar/isar/assets/71345/b919143b-f658-4fda-b81a-b543f193cd0b">
<img width="471" alt="image" src="https://github.com/isar/isar/assets/71345/69fc61d0-1000-4680-99e3-5f23bf8ca9aa">

Flutter 3.10.5
Dart 3.0.5
Isar v3.1.0+1

What kind of assistance can we offer you? This seems to be a major bug; for instance, in my app, this error has affected 20% of the users and each user has encountered it ten times.

**Comment by dennislean** [comment]
**OS:** macOS Ventura 13.4.1 (Apple M2)


**Flutter 3.10.0** • channel stable • https://github.com/flutter/flutter.git
Framework • revision 84a1e904f4 (3 months ago) • 2023-05-09 07:41:44 -0700
Engine • revision d44b5a94c9
Tools • **Dart 3.0.0** • DevTools 2.23.1

**Isar:** v3.1.0+1

**Error Message**
```
[VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: IsarError: Cannot open Environment: MdbxError (15): Block device required
```



**Sample code to reproduce the error:**
I'm using Isar example code (v3.1.0+1).

https://github.com/isar/isar/tree/3.1.0%2B1/examples/pub


**Steps to reproduce:**
1. Clone the repo and go to the example project.
2. Run pub get and build
3. Run the app without changing anything
4. Check the debug console

Note: If have markdown viewer problem, just remove it's usage and delete the file.

**Comment by simc** [comment]
@dennislean unfortunately it works perfectly fine on my M1 Pro with the same OS

**Comment by dennislean** [comment]
@simc 
Here's more info. For my case, the problem only happen when using iOS Simulator.

XCode: 14.3.1
Simulator: iPhone 14 Pro Max iOS 16.4

It works on physical phone on both Android and iPhone and Android emulator.

**Comment by simc** [comment]
I use exactly the same setup @dennislean 😕

**Comment by arlindiDev** [comment]
I also see the same error. I am building and running my app on M1 Apple macbook

**Comment by sarbazx** [comment]
Getting this error as well on iOS 17.0 Simulator. Any fixes?

**Comment by gsmlg** [comment]
Getting this error as well on macos x86_64 iOS 17.0 Simulator.

**Comment by jfcuylits** [comment]
@jackstefansky 
Did you find a solution for this problem?
As it seems to occurs on a significant % of devices, I wonder if I should continue with Isar or shift to another DB.

**Comment by johnstef99** [comment]
Managed to fix it by compiling isar rust and c dependencies locally, the bug is in the c code of mdbx lib. I opened [an issue there](https://gitflic.ru/project/erthink/libmdbx/issue/16).

**Comment by iblowmymind** [comment]
Found the cause of the issue for me, it was because I had Autodesk Flame installed at one point which requires some folders to be exported via NFS, and NFS exports makes the filesystem treated as MNT_EXPORTED. Disabling NFS via editing out /opt/Autodesk from /etc/exports and running `sudo nfsd disable` was fine for me because I don't use Flame anymore but for other workflows requiring NFS exports  it is technically a workaround, and an issue with libmdbx.
